apps/prairielearn/src/pages/courseSyncs/courseSyncs.js

```diff
@@ -24,84 +25,60 @@ router.get(
       throw new HttpStatusError(403, 'Access denied (must be course editor)');
     }
 
-    const jobSequencesResult = await sqldb.queryAsync(sql.select_sync_job_sequences, {
-      course_id: res.locals.course.id,
-    });
-    res.locals.job_sequences = jobSequencesResult.rows;
+    const jobSequences = await queryRows(
+      sql.select_sync_job_sequences,
+      { course_id: res.locals.course.id },
+      JobSequenceRowSchema,
+    );
 
-    const result = await sqldb.queryAsync(sql.question_images, {
-      course_id: res.locals.course.id,
-    });
-    res.locals.images = result.rows;
-    res.locals.imageSyncNeeded = false;
+    const images = await queryRows(
+      sql.question_images,
+      { course_id: res.locals.course.id },
+      ImageRowSchema,
+    );
 
     if (config.cacheImageRegistry) {
       const ecr = new ECR(makeAwsClientConfig());
-      await async.each(res.locals.images, async (image) => {
-        var repository = new DockerName(image.image);
+      await async.eachLimit(images, 3, async (image) => {
+        const repository = new DockerName(image.image);
         image.tag = repository.getTag() || 'latest (implied)';
         // Default to get overwritten later
         image.pushed_at = null;
         image.imageSyncNeeded = false;
         image.invalid = false;
 
+        /** @type {import('@aws-sdk/client-ecr').DescribeImagesCommandOutput} */
         let data;
         try {
           data = await ecr.describeImages({
             repositoryName: repository.getRepository(),
+            imageIds: [{ imageTag: repository.getTag() }],
```

Comment:
I believe you need this:

```suggestion
            imageIds: [{ imageTag: repository.getTag() ?? 'latest' }],
```

Reply:
This makes sense considering 
https://github.com/PrairieLearn/PrairieLearn/blob/d7b7b6ed345f8286a04cf5d1176c29fcedec6986/apps/prairielearn/src/pages/courseSyncs/courseSyncs.js#L44

Updated.